diff --git a/hierarchical_select.module b/hierarchical_select.module
index b9ace69..eb60598 100644
--- a/hierarchical_select.module
+++ b/hierarchical_select.module
@@ -1657,7 +1657,7 @@ function _hierarchical_select_hierarchy_generate($config, $selection, $required,
   // - the editability setting is enabled, and
   // - the hook is implemented (this is an optional hook), and
   // - the allowed_levels setting allows to create new items at this level.
-  if ($config['editability']['status']
+  if (!empty($config['editability']['status'])
         && module_hook($config['module'], 'hierarchical_select_create_item')
         && _hierarchical_select_create_new_item_is_allowed($config, 0)
   ) {
@@ -1698,7 +1698,7 @@ function _hierarchical_select_hierarchy_generate($config, $selection, $required,
     // - the editability setting is enabled, and
     // - the hook is implemented (this is an optional hook), and
     // - the allowed_levels setting allows to create new items at this level.
-    if ($config['editability']['status'] && module_hook($config['module'], 'hierarchical_select_create_item')) {
+    if (!empty($config['editability']['status']) && module_hook($config['module'], 'hierarchical_select_create_item')) {
       for ($depth = 1; $depth <= $max_depth; $depth++) {
         $item_type = t($config['editability']['item_types'][$depth]);
         $item_type = (!empty($item_type)) ? $item_type : t('item');
@@ -1723,7 +1723,7 @@ function _hierarchical_select_hierarchy_generate($config, $selection, $required,
       // - the hook is implemented (this is an optional hook), and
       // - the allowed_levels setting allows to create new items at this level.
       if ($depth > 0
-            && $config['editability']['status']
+            && !empty($config['editability']['status'])
             && module_hook($config['module'], 'hierarchical_select_create_item')
             && _hierarchical_select_create_new_item_is_allowed($config, $depth)
       ) {
@@ -1762,7 +1762,7 @@ function _hierarchical_select_hierarchy_generate($config, $selection, $required,
         // - the editability setting is enabled, and
         // - the hook is implemented (this is an optional hook), and
         // - the allowed_levels setting allows to create new items at this level.
-        if ($config['editability']['status']
+        if (!empty($config['editability']['status'])
               && module_hook($config['module'], 'hierarchical_select_create_item')
               && _hierarchical_select_create_new_item_is_allowed($config, $depth)
         ) {
@@ -1789,8 +1789,8 @@ function _hierarchical_select_hierarchy_generate($config, $selection, $required,
   // - an additional level is permitted by the max_levels setting
   // - the deepest item of the lineage is a valid item
   // NOTE: this uses an optional hook, so we also check if it's implemented.
-  if ($config['editability']['status']
-        && $config['editability']['allow_new_levels']
+  if (!empty($config['editability']['status'])
+        && !empty($config['editability']['allow_new_levels'])
         && ($config['editability']['max_levels'] == 0 || count($hierarchy->lineage) < $config['editability']['max_levels'])
         && module_invoke($config['module'], 'hierarchical_select_valid_item', end($hierarchy->lineage), $config['params'])
         && module_hook($config['module'], 'hierarchical_select_create_item')
